lib/chore/worker.rb

```diff
@@ -42,6 +42,24 @@ def expires_at
       @started_at + total_timeout
     end
 
+    def duplicate_work?(item)
+      # if we've got a duplicate, remove the message from the queue by not actually running and also not reporting any errors
+      payload = options[:payload_handler].payload(item.decoded_message)
+      
+      # only dedupe on the message id if we're getting work from an sqs consumer
```

Comment:
I don't understand this comment, is this so we don't fall over between SQS and the file consumer?

Reply:
yea, there's really no way for a duplicate-by-queue-id filesystem message to come in, so there's no sense in storing all of those in the cache

Reply:
I'll update that comment